### PR TITLE
Set template framework to latest

### PR DESCRIPTION
**Pull request by ILMTitan**
Update the template wizard to set template projects to the latest supported target framework version.

### CODE REVIEW

File: GoogleCloudExtension/GoogleCloudExtension/CloudExplorerSources/PubSub/TopicViewModelBase.cs [hunk 1]
```diff
@@ -61,7 +61,11 @@ internal abstract class TopicViewModelBase : TreeHierarchy, ITopicViewModelBase
 
         public ITopicItem Item { get; }
 
-        public event EventHandler ItemChanged;
+        event EventHandler ICloudExplorerItemSource.ItemChanged
```
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension/CloudExplorerSources/PubSub/TopicViewModelBase.cs]: This change seems unrelated to the rest of the PR.

File: GoogleCloudExtension/ProjectTemplates/GoogleAspnetWebApi/GoogleAspnetWebApi.csproj [hunk 2]
```diff
@@ -88,7 +88,6 @@
     <None Include="Controllers\HomeController.cs" />
     <None Include="Controllers\ValuesController.cs" />
     <None Include="Global.asax.cs" />
-    <None Include="Global.asax" />
```
Comment (ivannaranjo) [inline GoogleCloudExtension/ProjectTemplates/GoogleAspnetWebApi/GoogleAspnetWebApi.csproj]: This seems suspect, why removing this file?

File: GoogleCloudExtension/ProjectTemplates/GoogleAspnetMvc/GoogleAspnetMvc.csproj [hunk 3]
```diff
@@ -67,7 +67,6 @@
     <None Include="Controllers\HomeController.cs" />
     <None Include="Controllers\ManageController.cs" />
     <None Include="Global.asax.cs" />
-    <None Include="Global.asax" />
```
Comment (ivannaranjo) [inline GoogleCloudExtension/ProjectTemplates/GoogleAspnetMvc/GoogleAspnetMvc.csproj]: This seems suspect, why removing this file?

File: GoogleCloudExtension/ProjectTemplates/GoogleAspNetCoreMvc/Project.xproj [hunk 4]
```diff
@@ -10,7 +10,6 @@
     <RootNamespace>$safeprojectname$</RootNamespace>
     <BaseIntermediateOutputPath Condition="'$(BaseIntermediateOutputPath)'=='' ">.\obj</BaseIntermediateOutputPath>
     <OutputPath Condition="'$(OutputPath)'=='' ">.\bin\</OutputPath>
-    <TargetFrameworkVersion>v4.5.2</TargetFrameworkVersion>
```
Comment (ivannaranjo) [inline GoogleCloudExtension/ProjectTemplates/GoogleAspNetCoreMvc/Project.xproj]: This seems suspect, you remove it but not replacement for this line is offered? Was this line wrong before?

**Comment by ivannaranjo** [comment]
Looked at the change, and while it looks good, the problem with this approach is that you never actually update the dependencies in the project.json nor generate an updated <projec>.csproj. Meaning that yes, you change the target framework version, but the project is still using ASP.NET Core 1.0, when it should be using the version of ASP.NET that matches the target version.

We really need to have separate 1.1, and now 2.0, project templates.

File: GoogleCloudExtension/ProjectTemplates/GoogleAspnetMvc/GoogleAspnetMvc.csproj (hunk 3 above)
Comment (ILMTitan) [inline GoogleCloudExtension/ProjectTemplates/GoogleAspnetMvc/GoogleAspnetMvc.csproj]: This is a duplicate in the project. Removing this entry makes the file only show up once in the solution explorer.

**Comment by ILMTitan** [comment]
I am going to close this, as the framework chooser dialog is a better solution.
